File: src/test/java/com/github/daggerok/AppTest.java
package com.github.daggerok;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.context.support.GenericApplicationContext;

import static org.assertj.core.api.Assertions.assertThat;

/*
// Junit 4:
import org.junit.Test;

public class AppTest {

  @Test
  public void main() {
    GenericApplicationContext ctx = new AnnotationConfigApplicationContext(App.class);
    assertThat(ctx).isNotNull();
    App.main(new String[0]);
  }
}
*/

// Junit 5 (Jupiter):
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

@DisplayName("Junit 5 Test")
class AppTest {

  @Test
  void main() {
    GenericApplicationContext ctx = new AnnotationConfigApplicationContext(App.class);
    assertThat(ctx).isNotNull();
    App.main(new String[0]);
  }
}

File: src/test/java/com/github/daggerok/AppTest.kt
package com.github.daggerok

import assertk.assertThat
import assertk.assertions.isNotNull

/*
import org.junit.Test

class AppTest {

  @Test
  fun main() {
    val app = App()
    assertThat(app).isNotNull()
    com.github.daggerok.main()
  }
}
*/

import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Junit 5 test")
class AppTest {

  @Test
  fun main() {
    val app = App()
    assertThat(app).isNotNull()
    com.github.daggerok.main()
  }
}
